apidef/oas/oas.go

```diff
@@ -152,6 +152,21 @@ func (s *OAS) RemoveTykExtension() {
 	delete(s.Extensions, ExtensionTykAPIGateway)
 }
 
+func (s *OAS) Clone() (*OAS, error) {
+	oasInBytes, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+
+	var retOAS OAS
+	err = json.Unmarshal(oasInBytes, &retOAS)
+	if err != nil {
+		return nil, err
+	}
+
+	return &retOAS, nil
+}
+
```

Comment:
Test + benchmark please. Assert modification in returned value and that the original (receiver) value remains unmodified. Modify some nested pointer or map/slice value.